sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/spice_indexer.py

```diff
@@ -371,87 +371,72 @@ def index_pointing_data(s3_key: str):
-                | (models.PointingTable.repoint_start_utc.is_(None))
-                | (models.PointingTable.repoint_end_utc.is_(None))
-            )
-            .all()
-        ):
-            repoint_row = repoint_df[
-                repoint_df["repoint_id"] == pointing_entry.pointing_id
-            ]
-            if not repoint_row.empty:
-                next_row = repoint_df[
-                    repoint_df["repoint_id"] == pointing_entry.pointing_id + 1
-                ]
-
-                if not next_row.empty:
-                    pointing_entry.pointing_end_utc = pd.to_datetime(
-                        next_row.iloc[0]["repoint_end_utc"]
-                    )
-                    pointing_entry.repoint_start_utc = pd.to_datetime(
-                        next_row.iloc[0]["repoint_start_utc"]
-                    )
-                    pointing_entry.repoint_end_utc = pd.to_datetime(
-                        next_row.iloc[0]["repoint_end_utc"]
-                    )
-                else:
-                    pointing_entry.repoint_start_utc = None
-                    pointing_entry.repoint_end_utc = None
-                    pointing_entry.pointing_end_utc = None
-
-        # Filter repoint_id that's not in pointing_table
-        pointing_ids = session.query(models.PointingTable.pointing_id).all()
-        existing_ids = [id[0] for id in pointing_ids]
-
-        # Only process repoint_ids not already in the table
-        new_repoint_df = repoint_df[~repoint_df["repoint_id"].isin(existing_ids)]
-
-        # For each new repoint_id, calculate pointing_start_utc and pointing_end_utc
-        for _, row in new_repoint_df.iterrows():
-            repoint_id = row["repoint_id"]
-            try:
-                # Convert to datetime for SQLite compatibility
-                pointing_start_utc = pd.to_datetime(row["repoint_end_utc"])
-                next_row = repoint_df[repoint_df["repoint_id"] == repoint_id + 1]
-
-                if not next_row.empty:
-                    pointing_end_utc = pd.to_datetime(
-                        next_row.iloc[0]["repoint_end_utc"]
-                    )
-                    repoint_start_utc = pd.to_datetime(
-                        next_row.iloc[0]["repoint_start_utc"]
-                    )
-                    repoint_end_utc = pd.to_datetime(
-                        next_row.iloc[0]["repoint_end_utc"]
-                    )
-                else:
-                    repoint_start_utc = None
-                    repoint_end_utc = None
-                    pointing_end_utc = None
-
-            except Exception as e:
-                logger.error(
-                    f"Error calculating pointing UTCs for repoint_id {repoint_id}: {e}"
-                )
-                continue
-
-            params = {
+    max_repoint_id = repoint_df["repoint_id"].max()
+    repoint_records = []
+
+    for repoint_id in repoint_df["repoint_id"].values:
```

Comment:
Iterate over all but the final row. Does astype(int) work here so you don't have to cast below?
```suggestion
    for i_row, repoint_id in enumerate(repoint_df["repoint_id"].values[:-1].astype(int)):
```